### PR TITLE
UTOPIA-218: ReactJS Dockerfile

### PR DESCRIPTION
**Pull request by Bottle7**
# Description

This PR includes the following proposed change(s):

- Adds Dockerfile to frontend
- Edits path to frontend in the docker-compose

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Spun up the frontend container using docker-compose and changed the App.css file to App.scss to test the sass was working

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [x] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [x] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

### CODE REVIEW

File: src/frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#############################################################################################
+###                                 BUILD ARTIFACT                                        ###
+#############################################################################################
```
Comment (akroon3r) [inline src/frontend/Dockerfile:3]: We can remove if we're not using a multi stage build strategy with NGINX

File: src/frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+WORKDIR /app
+
+# Copy in build artifacts, build project dependencies
+
+COPY src/frontend/package*.json ./
+
+RUN npm install
+
+# NEED THIS FOR SASS TO BE AVAILABLE IN CONTAINER
```
Comment (akroon3r) [inline src/frontend/Dockerfile:14]: Please don't yell at me. 😢

File: src/frontend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+COPY src/frontend/. .
+
+RUN npm run build
+
+WORKDIR /app/dist
+
+CMD ["serve", "-p", "8080"]
+
+# THIS IS FOR DEBUGGING
+# CMD ["sleep", "3600"]
```
Comment (akroon3r) [inline src/frontend/Dockerfile:29]: Remove if not needed please.

**Review by akroon3r — COMMENTED** [review]
Nice job @Bottle7  @antsand. Just a few comments to follow-up on. Great job tag teaming this.

File: src/frontend/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+FROM node:lts-alpine as build
```
Comment (kushal-arora-fw) [inline src/frontend/Dockerfile:1]: Can we please make it a more specific version like `node:18.9-alpine` to avoid any unforeseen breaks in the future?

**Review by antsand — APPROVED** [review]
Looks good. For the next commit change src/index.css to src/index.scss

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]

File: src/frontend/Dockerfile (hunk 4 above)
Comment (akroon3r) [reply src/frontend/Dockerfile:1]: That's a great point Kushal. Anthony and I ran into issues a month ago using this exact same image. 

Good catch.

Comment (antsand) [reply src/frontend/Dockerfile:1]: good point.

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**